web/src/actions/index.js

```diff
@@ -120,8 +120,12 @@ export function submitSentences(language, sentences, source) {
     const db = new WebDB(state.app.username, state.app.password);
     const results = await db.submitSentences(language, sentences, source);
     dispatch(submitSentencesSuccess(results.sentences.slice(0)));
+    dispatch( sendAddLanguage() );
     const errorsWithSentenceInfo = results.errors.filter((error) => error.sentence);
     dispatch(submitSentencesFailureSingle(errorsWithSentenceInfo));
+    if( !state.app.languages.includes( language ) ) {
+      dispatch(addLanguageSuccess(language));
+    }
```

Comment:
While this adds to the profile, it won't save it in the database. You can check this by logging out and back in. Instead of having these two calls here, I'd suggest to only call `addLanguage(language)` at the end of the function which will dispatch the right messages as well as save it to the database.

Reply:
While testing the working of the app, I should have tested by logging out and logging back in.
I am sorry and will work to resolve this issue.
You should have a PR by day after tomorrow.

Reply:
Absolutely no need to be sorry :)